sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1206,7 +1207,7 @@ case class DatetimeSub(
     interval: Expression,
     child: Expression) extends RuntimeReplaceable {
   override def toString: String = s"$start - $interval"
-  override def sql: String = s"${start.sql} - ${interval.sql}"
+  override def sql: String = s"${toPrettySQL(start)} - ${toPrettySQL(interval)}"
```

Comment:
This means `RuntimeReplaceable.sql` always use pretty sql, while other expressions only use pretty sql when the caller side needs to (by calling `toPrettySQL`).

Is it possible to make them consistent? Like make `start` and `interval` as `innerChildren` and handle them in `usePrettyExpression`.

Reply:
OK, let me see

Reply:
OK, let me see